src/cpp/configuration/DDSRouterConfiguration.cpp

```diff
@@ -18,140 +18,136 @@
+std::set<std::shared_ptr<FilterTopic>> DDSRouterConfiguration::allowlist() const noexcept
 {
-    return generic_get_topic_list_(ALLOWLIST_TAG);
+    return allowlist_;
 }
 
-std::list<std::shared_ptr<FilterTopic>> DDSRouterConfiguration::blocklist() const
+std::set<std::shared_ptr<FilterTopic>> DDSRouterConfiguration::blocklist() const noexcept
 {
-    return generic_get_topic_list_(BLOCKLIST_TAG);
+    return blocklist_;
 }
 
-std::list<ParticipantConfiguration> DDSRouterConfiguration::participants_configurations() const
+std::set<std::shared_ptr<RealTopic>> DDSRouterConfiguration::builtin_topics() const noexcept
 {
-    std::list<ParticipantConfiguration> result;
+    return builtin_topics_;
+}
 
-    for (YAML::const_iterator participant_it = raw_configuration_.begin();
-            participant_it != raw_configuration_.end();
-            ++participant_it)
-    {
-        std::string id_str = participant_it->first.as<std::string>();
+std::set<std::shared_ptr<ParticipantConfiguration>> DDSRouterConfiguration::participants_configurations() const noexcept
+{
+    return participants_configurations_;
+}
 
-        // Check if it is a valid name for a participant
-        if (!ParticipantId::is_valid_id(id_str))
+bool DDSRouterConfiguration::is_valid(utils::Formatter& error_msg) const noexcept
+{
+    // Check Allow list topics
+    for (std::shared_ptr<FilterTopic> topic : allowlist_)
+    {
+        if (!topic)
         {
-            continue;
+            logError(DDSROUTER_CONFIGURATION, "Invalid ptr in allowlist topics.");
+            error_msg << "nullptr Filter Topic in allowlist.";
+            return false;
         }
 
-        // Add new Participant with its configuration
-        // NOTE: There will not be repeated Participant Ids as in a yaml the keys are unique
-        result.push_back(ParticipantConfiguration(ParticipantId(id_str), participant_it->second));
+        if (!topic->is_valid())
+        {
+            error_msg << "Invalid Filter Topic " << topic << " in allowlist.";
+            return false;
+        }
     }
 
-    return result;
-}
-
-std::set<RealTopic> DDSRouterConfiguration::real_topics() const
-{
-    std::set<RealTopic> result;
-
-    for (const std::shared_ptr<FilterTopic>& topic : generic_get_topic_list_(ALLOWLIST_TAG))
+    // Check Block list topics
+    for (std::shared_ptr<FilterTopic> topic : blocklist_)
     {
-        if (RealTopic::is_real_topic(topic->topic_name(), topic->topic_type()))
+        if (!topic)
+        {
+            logError(DDSROUTER_CONFIGURATION, "Invalid ptr in blocklist topics.");
+            error_msg << "nullptr Filter Topic in blocklist.";
+            return false;
+        }
+
+        if (!topic->is_valid())
         {
-            result.emplace(RealTopic(topic->topic_name(), topic->topic_type(), topic->topic_with_key()));
+            error_msg << "Invalid Filter Topic " << topic << " in blocklist.";
```

Comment:
The idea was to show an error when something that should not happen, happened. (not because C++ weird behaviour, but because user bad practice).
And in cases where the error is just that the input is not valid, not show an error.

But I agree that we should study how to manage error messages and where to add them.